webapp/src/app/components/widgets/event-widget/event-registry/abstract-event.ts

```diff
@@ -93,6 +164,38 @@ export abstract class AbstractEvent<T extends EventType> {
 	protected getColorRectangle(color: string): string {
 		return `<span style="background-color: ${this.sanitize(color)};">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>`;
 	}
+
+	protected getProcessedText(langLabel: Label): string {
```

Comment:
There is, I just didn't want to introduce an external library where it wasn't necessary (or without permission), but if you would like I absolutely use that library.

Reply:
I just tried to implement [cc-textparser](https://github.com/redstonekasi/cc-textparser/) into the editor and whenever it reached invalid commands (even just as simple as typing in `\c` before typing in the whole command) it completely locked up the editor. I suggest not using it now unless that can be resolved - and I don't think I used it incorrectly, either. I feel in its current state, it is not meant for live text editing.

While I think using that textparser would be an improvement, I would suggest saving implementing it until later - whenever those issues with it get resolved.

Reply:
@redstonekasi Any ideas?

Reply:
Gimme a few minutes, I'll make a diff so you can easily see what is happening (and maybe see if I did something wrong).

Reply:
Here, I uploaded it to discord, so you can check it out: https://ptb.discord.com/channels/382339402338402315/383264360518123520/1142964077518397460

Reply:
@EL20202 Here's a parser that handles invalid input: https://github.com/dmitmel/crosscode-translateinator/blob/c325cdad7b6f6ca40c9a73e6a98527a385a5aa4b/src/crosscode_markup.ts

Reply:
I cannot recommend using cc-textparser right now since it was originally meant as a port of the actual CrossCode source without regard to actual usability and performance. Especially in the context of an editor such as this, not having fault tolerant parsing (is that the right term?) is not nice at all.
Not sure what is causing you issues in that example you posted on Discord since it is missing most of the relevant code.